release/scripts/workspaces.mjs

```diff
@@ -0,0 +1,556 @@
+  }
+
+  async release() {
+    if (this.options.publish === false) return;
+
+    // creating a stable object that is shared across all package publishes
+    // this ensures that we don't accidentally prompt multiple times (e.g. once
+    // per package) due to loosing the otp value after each `this.publish` call
+    const otp = {
+      value: this.options.otp,
+    };
+
+    const tag = this.getContext('distTag');
+    const task = async () => {
+      await this.eachWorkspace(async (workspaceInfo) => {
+        await this.publish({ tag, workspaceInfo, otp });
+      });
+    };
+
+    await this.step({ task, label: 'npm publish', prompt: 'publish' });
+  }
+
+  async afterRelease() {
+    let workspaces = this.getWorkspaces();
+
+    workspaces.forEach((workspaceInfo) => {
+      if (workspaceInfo.isReleased) {
+        this.log.log(`🔗 ${this.getReleaseUrl(workspaceInfo)}`);
+      }
+    });
+  }
+
+  _buildReplacementDepencencyVersion(existingVersion, newVersion) {
+    let firstChar = existingVersion[0];
+
+    // preserve existing floating constraint
+    if (['^', '~'].includes(firstChar)) {
+      return `${firstChar}${newVersion}`;
+    }
+
+    return newVersion;
+  }
+
+  _updateDependencies(pkgInfo, newVersion) {
+    const { isDryRun } = this.config;
+    const workspaces = this.getWorkspaces();
+    const { pkg } = pkgInfo;
+
+    const updateDependencies = (dependencyType) => {
+      let dependencies = pkg[dependencyType];
+
+      if (dependencies) {
+        for (let dependency in dependencies) {
+          if (workspaces.find((w) => w.name === dependency)) {
+            const existingVersion = dependencies[dependency];
+            const replacementVersion = this._buildReplacementDepencencyVersion(
+              existingVersion,
+              newVersion
+            );
+
+            this.log.exec(
+              `\t${dependencyType}: \`${dependency}\` -> ${replacementVersion} (from ${existingVersion})`
+            );
+
+            if (!isDryRun) {
+              dependencies[dependency] = replacementVersion;
+            }
+          }
+        }
+      }
+    };
+
+    updateDependencies('dependencies');
+    updateDependencies('devDependencies');
+    updateDependencies('optionalDependencies');
+    updateDependencies('peerDependencies');
+  }
+
+  _formatPublishMessage(distTag, packageNames) {
+    const messages = [
+      'Preparing to publish:',
+      ...packageNames.map((name) => `    ${name}${distTag === 'latest' ? '' : `@${distTag}`}`),
+      '  Publish to npm:',
+    ];
+
+    return messages.join('\n');
+  }
+
+  async isRegistryUp() {
+    const registry = this.getRegistry();
+
+    try {
+      await this.exec(`npm ping --registry ${registry}`);
+
+      return true;
+    } catch (error) {
+      if (/code E40[04]|404.*(ping not found|No content for path)/.test(error)) {
+        this.log.warn('Ignoring unsupported `npm ping` command response.');
+        return true;
+      }
+      return false;
```

Comment:
```suggestion

      return false;
```

Reply:
This code was copied as it is from `https://github.com/release-it-plugins/workspaces`. Since we should ideally wait till the PR we sent them is merged, this code has been added to this repo temporarily. So, I don't think we need to format their code.